populator-machinery/controller.go

```diff
@@ -108,13 +114,65 @@ type controller struct {
 	recorder             record.EventRecorder
 	referenceGrantLister referenceGrantv1beta1.ReferenceGrantLister
 	referenceGrantSynced cache.InformerSynced
+	useProviderImpl      bool
+	populate             func(context.Context, *PopulatorParams) error
+	populateComplete     func(context.Context, *PopulatorParams) (bool, error)
+}
+
+type VolumePopulatorConfig struct {
+	MasterURL     string
+	Kubeconfig    string
+	ImageName     string
+	HttpEndpoint  string
+	MetricsPath   string
+	Namespace     string
+	Prefix        string
+	Gk            schema.GroupKind
+	Gvr           schema.GroupVersionResource
+	MountPath     string
+	DevicePath    string
+	PopulatorArgs func(bool, *unstructured.Unstructured) ([]string, error)
+	// Boolean flag which determines whether or not to use cloud provider specific data populate functions
+	UseProviderImpl bool
```

Comment:
Do you need this boolean? Can it be determined by whether or not Populate is set?

Reply:
I removed this boolean flag, use PopulatorArgs!=nil and Populate!=nil instead. If user passes in PopulatorArgs!=nil and  Populate!=nil at the same time, they can have populator pod and provider function at the same time. We will need more tests for this case to make sure there is no conflict when both are used.